docs/SIL.rst

```diff
@@ -3102,6 +3102,9 @@ The ``dynamic_lifetime`` attribute specifies that the initialization and
 destruction of the stored value cannot be verified at compile time.
 This is the case, e.g. for conditionally initialized objects.
 
+The optional ``lexical`` attribute specifies that the operand corresponds to a
```

Comment:
alloc_stack does not have an operand.

How is `lexical` different to `debug-var-attr`? Doesn't alloc_stack correspond to a variable if `debug-var-attr` is present?

Reply:
Thanks!  Fixed at https://github.com/apple/swift/pull/39484 .

Is the `debug-var-attr` always present?  If it is, @atrick may know the answer.  At worst, this might be duplicative and we'll need to tweak the change to Mem2Reg.

Reply:
As a general rule, debug information is not supposed to affect semantics. If we decide to reuse the same information for both purposes, then we need to specify in the class and API that it is not simply for debug information, audit, and verify the implementation.

Reply:
Still, I would prefer to not have redundant information in the instruction attributes.

Reply:
@adrian-prantl do you know in which cases `debug-var-attr` is present?

Reply:
`debug-var-attr` is present iff the alloc_stack is the only storage location for a (part of a) source variable. It is there regardless of whether -g is on or not. There can be alloc_stack instructions that don't have a debug-var-attr, in purely compiler-generated code (thunks, etc).